packages/dev/core/src/FlowGraph/Blocks/Event/flowGraphReceiveCustomEventBlock.ts

```diff
@@ -0,0 +1,43 @@
+import type { Observer } from "../../../Misc/observable";
+import type { FlowGraphContext } from "../../flowGraphContext";
+import { FlowGraphEventBlock } from "../../flowGraphEventBlock";
+import type { FlowGraphCustomEvent } from "../../flowGraphCustomEvent";
+import type { Nullable } from "../../../types";
+import { Tools } from "../../../Misc";
+import type { FlowGraphDataConnection } from "../../flowGraphDataConnection";
+import { RichTypeAny } from "../../flowGraphRichTypes";
+
+export interface IFlowGraphReceiveCustomEventBlockParameters {
+    eventId: string;
+}
+
+export class FlowGraphReceiveCustomEventBlock extends FlowGraphEventBlock {
+    private _params: IFlowGraphReceiveCustomEventBlockParameters;
+    private _eventObserver: Nullable<Observer<FlowGraphCustomEvent>>;
+
+    /**
+     * Output connection: The data sent with the event.
+     */
+    public eventData: FlowGraphDataConnection<any>;
+
+    constructor(params: IFlowGraphReceiveCustomEventBlockParameters) {
+        super();
+        this._params = params;

```

Comment:
Save a few lines of code:

```suggestion
    constructor(private _params: IFlowGraphReceiveCustomEventBlockParameters) {
        super();
```

And remove the declaration in line 15